lib/silo/commands/file_push.rb

```diff
@@ -40,14 +40,12 @@ def run
 
         source = args[0]
 
-        if args[1]&.match(/^[^:]*:[^:]*$/)
-          silo_name, dest = args[1].split(":").map(&:to_s)
-        else
-          silo_name = Silo.default
-          dest = args[1] || ''
-        end
-
-        dest ||= ''
+        # standardize args[1] to [silo_name, dest]
+        split_args = args[1]&.split(":", 2).map(&:to_s) || ['']
```

Comment:
This doesn't work. If `args[1]` is `nil`, you will get an error upon attempting to call `map` on it (`split` will return `nil` because of the nilsafe operator).

Also, the following test looks incorrect:

```ruby
2.7.1 :017 > def test(str)
2.7.1 :018 >   x = str&.split(':', 2).map(&:to_s) || ['']
2.7.1 :019 >   x.tap { |f| f.unshift('jack-silo') if x.size == 1 }
2.7.1 :020 > end
 => :test
2.7.1 :024 > test(':f')
 => ["", "f"]
```

`:f` is an invalid input, as it contains no silo name and therefore a filename with a colon in it`.